docs/streams/developer-guide/dsl-api.html

```diff
@@ -2374,6 +2387,328 @@ <h4><a id="streams_concepts_globalktable" href="#streams_concepts_globalktable">
+                                          key are ignored and do not trigger the join.</li>
+                                        <li>Input records with a <code class="docutils
+                                            literal"><span class="pre">null</span></code>
+                                          value are interpreted as <em>tombstones</em>
+                                          for the corresponding key, which indicate the
+                                          deletion of the key from the table. Tombstones
+                                          do not trigger the join. When an input
+                                          tombstone is received, then an output
+                                          tombstone is forwarded directly to the join
+                                          result KTable if required (i.e. only if the
+                                          corresponding key actually exists already in
+                                          the join result KTable).</li>
+                                      </ul>
+                                    </div>
+                                  </blockquote>
+                                </li>
+                                <li>
+                                  <p class="first">For each input record on the left
+                                    side that does not have any match on the right side,
+                                    the <code class="docutils literal"><span
+                                        class="pre">ValueJoiner</span></code> will be
+                                    called with <code class="docutils literal"><span
+                                        class="pre">ValueJoiner#apply(leftRecord.value,</span>
+                                      <span class="pre">null)</span></code>; this
+                                    explains the row with timestamp=7 &amp; 8 in the
+                                    table below, which lists <code class="docutils
+                                      literal"><span class="pre">(q,10,null) and (r,10,null)</span></code>
+                                    in the LEFT JOIN column.</p>
+                                </li>
+                              </ul>
+                              <p class="last">See the semantics overview at the bottom
+                                of this section for a detailed description.</p>
+                            </td>
+                          </tr>
+                        </tbody>
+                      </table>
+                      <p><strong>Semantics of table-table foreign-key joins:</strong>
+                        The semantics of the various table-table foreign-key join
+                        variants are explained below. All records are processed in
+                        timestamp order. The key is shown alongside the value for each
```

Comment:
`timestamp order` -- that sounds miss leading. Per partition, records are always processed in offset order. Cross-partition, we do timestamp synchronization though. Can we refine this? Or maybe remove completely?

Reply:
I had taken this from other KTable-KTable join sections. I will elaborate on it in a clearer way.

Reply:
Okay I tried to make it clearer. LMK what you think.